crates/router/src/services/authentication.rs

```diff
@@ -684,6 +698,43 @@ where
                 merchant_id: payload.merchant_id.clone(),
                 org_id: payload.org_id,
                 role_id: payload.role_id,
+                profile_id: payload.profile_id,
+            },
+            AuthenticationType::MerchantJwt {
+                merchant_id: payload.merchant_id,
+                user_id: Some(payload.user_id),
+            },
+        ))
+    }
+}
+
+#[cfg(feature = "olap")]
+#[async_trait]
+impl<A> AuthenticateAndFetch<AuthenticationDataWithMultipleProfiles, A> for JWTAuth
+where
+    A: SessionStateInfo + Sync,
+{
+    async fn authenticate_and_fetch(
+        &self,
+        request_headers: &HeaderMap,
+        state: &A,
+    ) -> RouterResult<(AuthenticationDataWithMultipleProfiles, AuthenticationType)> {
+        let payload = parse_jwt_payload::<A, AuthToken>(request_headers, state).await?;
+        if payload.check_in_blacklist(state).await? {
+            return Err(errors::ApiErrorResponse::InvalidJwtToken.into());
+        }
+
+        let permissions = authorization::get_permissions(state, &payload).await?;
+        authorization::check_authorization(&self.0, &permissions)?;
+
+        Ok((
+            AuthenticationDataWithMultipleProfiles {
+                org_id: payload.org_id,
+                merchant_id: payload.merchant_id.clone(),
+                profile_id: payload
+                    .profile_id
+                    .map(|profile_id| vec![profile_id])
+                    .unwrap_or(vec![]),
```

Comment:
Are we not supposed to validate this merchant_id and profile_id list?

Reply:
what kind of validation?

Reply:
Like check if the MerchantAccount is present in DB, And fetch and return the record.
And also check if all profiles belong to the given merchant.

Reply:
No, we don't have to.

Since this JWT is issued and signed by us we can always say that this JWT will have valid profile_ids.

We can do it for APIKey auth where the profile_ids are sent by merchant themselves.